src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FC, ReactNode, RefObject } from 'react';
```

Comment:
Feels like we should allow prop spreading down to the DialogOverlay to cover any options that are exposed by reach.